src/hugr/view.rs

```diff
@@ -227,3 +226,43 @@ where
         self.deref()
     }
 }
+
+/// Iterator over the links of a port
+#[derive(Clone)]
+pub struct PortLinks<'a> {
+    hugr: &'a Hugr,
+    links: crate::hugr::multiportgraph::PortLinks<'a>,
+}
+
+impl<'a> Iterator for PortLinks<'a> {
+    type Item = (Node, Port);
```

Comment:
Wonder if this is worth a more general type definition (not in this PR though).

Reply:
The iterator story is quite cumbersome. Hopefully we can clean it up in the future.